File: todo-app/src/types/project.ts
export interface Project {
  id: number;
  name: string;
  icon: string;
  description: string;
  createdAt: string;
}

export interface CreateProjectInput {
  name: string;
  icon: string;
  description: string;
}

export interface UpdateProjectInput {
  name?: string;
  icon?: string;
  description?: string;
}

File: backend/app/models/project.py
from sqlalchemy import Column, String, Text, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
import uuid
from ..core.database import Base


class Project(Base):
    __tablename__ = "projects"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String(100), nullable=False, unique=True)
    description = Column(Text)
    owner_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"))
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # Relationship to owner
    owner = relationship("User", back_populates="projects")

    # Relationships to keywords and test data
    keywords = relationship("Keyword", back_populates="project")
    test_data = relationship("TestData", back_populates="project")